sklearn/kernel_ridge/krc.py

```diff
@@ -0,0 +1,132 @@
+
+    kernel : string or callable, default="rbf"
+        Kernel mapping used internally. A callable should accept two arguments
+        and the keyword arguments passed to this object as kernel_params, and
+        should return a floating point number.
+
+    gamma : float, default=None
+        Gamma parameter for the RBF, laplacian, polynomial, exponential chi2
+        and sigmoid kernels. Interpretation of the default value is left to
+        the kernel; see the documentation for sklearn.metrics.pairwise.
+        Ignored by other kernels.
+
+    degree : float, default=3
+        Degree of the polynomial kernel. Ignored by other kernels.
+
+    coef0 : float, default=1
+        Zero coefficient for polynomial and sigmoid kernels.
+        Ignored by other kernels.
+
+    kernel_params : mapping of string to any, optional
+        Additional parameters (keyword arguments) for kernel function passed
+        as callable object.
+
+    Attributes
+    ----------
+    dual_coef_ : array, shape = [n_samples] or [n_samples, n_targets]
+        Representation of weight vector(s) in kernel space
+
+    X_fit_ : {array-like, sparse matrix}, shape = [n_samples, n_features]
+        Training data, which is also required for prediction
+
+    classes_ : array of shape = [n_classes] or a list of such arrays
+        The classes labels (single output problem),
+        or a list of arrays of class labels (multi-output problem).
+
+    References
+    ----------
+    * Kevin P. Murphy
+      "Machine Learning: A Probabilistic Perspective", The MIT Press
+      chapter 14.4.3, pp. 492-493
+
+    * An, Senjian, Wanquan Liu, and Svetha Venkatesh.
+      "Face recognition using kernel ridge regression." Computer Vision
+      and Pattern Recognition, 2007. CVPR'07. IEEE Conference on. IEEE,
+      2007.
+
+    See also
+    --------
+    sklearn.linear_model.Ridge:
+        Linear ridge regression.
+    sklearn.kernel_ridge.KernelRidge:
+        Kernel Ridge implemented for regressions.
+
+    Examples
+    --------
+    >>> from sklearn.kernel_ridge import KernelRidgeClassifier
+    >>> X = [[-2, -1], [-1, -1], [-1, -2], [1, 1], [1, 2], [2, 1]]
+    >>> y = [1, 1, 1, 2, 2, 2]
+    >>> clf = KernelRidgeClassifier()
+    >>> clf.fit(X, y) # doctest: +NORMALIZE_WHITESPACE
+    KernelRidgeClassifier(alpha=0.1, coef0=1, degree=3,
+                          gamma=None, kernel='rbf',
+                          kernel_params=None)
+    """
+    def __init__(self, alpha=0.1, kernel="rbf", gamma=None, degree=3, coef0=1,
+                 kernel_params=None):
+        super(KernelRidgeClassifier,
+              self).__init__(alpha=alpha,
+                             kernel=kernel,
+                             gamma=gamma,
+                             degree=degree,
+                             coef0=coef0,
+                             kernel_params=kernel_params)
+
+    def decision_function(self, X):
```

Comment:
To avoid repeating the implementation, you should call super(...).predict here, and use method overloading to define predict in KRC

Reply:
Done e12d1e3

Reply:
Done ( e12d1e38daba4ee92b5bbb2c10d431b934d82b8d )